src/Umbraco.Web/Redirects/RedirectUrlManagementController.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Net.Http;
+using System.Text;
+using System.Web;
+using System.Web.Http;
+using System.Xml;
+using Umbraco.Core.Configuration;
+using Umbraco.Core.Services;
+using Umbraco.Web.WebApi;
+using File = System.IO.File;
+
+namespace Umbraco.Web.Redirects
+{
+    public class RedirectUrlManagementController : UmbracoAuthorizedApiController
+    {
+        //add paging
+        [HttpGet]
+        public RedirectUrlSearchResult SearchRedirectUrls(string searchTerm, int page = 0, int pageSize = 10)
+        {
+            var searchResult = new RedirectUrlSearchResult { UrlTrackerDisabled = UmbracoConfig.For.UmbracoSettings().WebRouting.DisableRedirectUrlTracking };
+            var redirectUrlService = Services.RedirectUrlService;
+            var resultCount = 0L;
+
+            var redirects = string.IsNullOrWhiteSpace(searchTerm) 
+                ? redirectUrlService.GetAllRedirectUrls(page, pageSize, out resultCount)
+                : redirectUrlService.SearchRedirectUrls(searchTerm, page, pageSize, out resultCount);
+
+            searchResult.SearchResults = redirects;
+            searchResult.TotalCount = resultCount;
+            searchResult.CurrentPage = page;
+            searchResult.PageCount = ((int)resultCount + pageSize - 1) / pageSize;
+
+            searchResult.HasSearchResults = resultCount > 0;
+            searchResult.HasExactMatch = (resultCount == 1);
+            return searchResult;
+
+        }
+
+        [HttpGet]
+        public HttpResponseMessage GetPublishedUrl(int id)
+        {
+            var publishedUrl = "#";
+            if (id > 0)
+                publishedUrl = Umbraco.Url(id);
+
+            return new HttpResponseMessage { Content = new StringContent(publishedUrl, Encoding.UTF8, "text/html") };
+
+        }
+
+        [HttpPost]
+        public IHttpActionResult DeleteRedirectUrl(Guid id)
+        {
+            var redirectUrlService = Services.RedirectUrlService;
+            redirectUrlService.Delete(id);
+            return Ok();
+        }
+
+        [HttpPost]
+        public IHttpActionResult ToggleUrlTracker(bool disable)
+        {
+            var configFilePath = HttpContext.Current.Server.MapPath("~/config/umbracoSettings.config");
```

Comment:
Generally rule: Never ever use `HttpContext.Current` - there is no need for this, anytime a singleton is used think if this is a 'bad practice'. This is not testable and the MapPath function is available on `HostingEnvironment.MapPath` (still not ideal) but you can get access to the `HttpContext` in webapi controllers by using `TryGetHttpContext` method